include/sampleflow/consumers/autocovariance_dim_1.h

```diff
@@ -0,0 +1,255 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef SAMPLEFLOW_CONSUMERS_AUTOCOVARIANCEDIM1_H
+#define SAMPLEFLOW_CONSUMERS_AUTOCOVARIANCEDIM1_H
+
+#include <sampleflow/consumer.h>
+#include <sampleflow/types.h>
+#include <mutex>
+
+#include <boost/numeric/ublas/matrix.hpp>
+
+
+namespace SampleFlow
+{
+  namespace Consumers
+  {
+    /**
+     * A Consumer class that implements computing the running sample autocovariance function:
+     * \hat\gamma(k)=\frac{1}{n}\sum_{t=1}^{n-k}{(x_{t+k}-\bar{x})(x_{t}-\bar{x})}
+     *
+     * ### Threading model ###
+     *
+     * The implementation of this class is thread-safe, i.e., its
+     * consume() member function can be called concurrently and from multiple
+     * threads.
+     *
+     *
+     * @tparam InputType The C++ type used for the samples $x_k$. In
+     *   order to compute covariances, the same kind of requirements
+     *   have to hold as listed for the Covariance class.
```

Comment:
But actually, you're only considering the 1d case here, so the requirements on the data type may be different. You might want to think about what kind of type `InputType` can actually be.